buildrunner/steprunner/tasks/push.py

```diff
@@ -95,74 +82,86 @@ def __init__(self, step_runner, config, commit_only=False):
         else:
             self._repos = [self._get_repo_definition(config)]
 
-    def run(self, context):
-        # first see if a run task produced an image (via a post-build config)
-        if 'run-image' in context:
-            image_to_use = context.get('run-image')
-        # next see if there was a run task, committing the end state of the
-        # container as the image to use
-        elif 'run_runner' in context:
-            image_to_use = context['run_runner'].commit(self.step_runner.log)
-        # finally see if we have an image from a build task
-        else:
-            image_to_use = context.get('image', None)
-
-        # validate we have an image
-        if not image_to_use:
-            raise BuildRunnerProcessingError(
-                'Cannot find an image to tag/push from a previous task'
-            )
-        self.step_runner.log.write(
-            f'Using image {image_to_use} for tagging\n'
-        )
+    def run(self, context):  # pylint: disable=too-many-branches
 
         # Determine internal tag based on source control information and build number
         # Note: do not log sanitization as it would need to happen almost every time
         default_tag = sanitize_tag(self.step_runner.build_runner.build_id)
 
-        # add the image to the list of generated images for potential cleanup
-        self.step_runner.build_runner.generated_images.append(image_to_use)
-
-        for repo in self._repos:
-            # Always add default tag
-            repo.tags.append(default_tag)
+        # Tag multi-platform images
+        if self.step_runner.multi_platform.is_multiplatform(self.get_unique_build_name()):
+            for repo in self._repos:
+                # Always add default tag
+                repo.tags.append(default_tag)
+                self.step_runner.multi_platform.tag_single_platform(name=self.get_unique_build_name(),
```

Comment:
Personally I feel like this kind of parameter indentation can be hard to read, especially with very long prefixes. I much prefer something like this:

```
self.step_runner.multi_platform.tag_single_platform(
    name=self.get_unique_build_name(),
    tags=repo.tags,
    dest_name=repo.repository,
)
```

I'm not asking you to change this, I just thought I should mention it as I think it's much easier to parse what's going on since you don't have to read across the entire line for the indent.

Reply:
I like your suggestion better.

Reply:
done